distribution_strategy/parameter_server_training/resnet_cifar_ps_strategy.py

```diff
@@ -0,0 +1,289 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+# Lint as: python3
+"""ResNet Cifar + ParameterServerStrategy example.
+"""
+import os
+from datetime import datetime
+from absl import app
+from absl import flags
+from absl import logging
+import portpicker
+import tensorflow as tf
+from tensorflow.python.distribute import parameter_server_strategy_v2
+from tensorflow_models.official.benchmark.models import cifar_preprocessing
+from tensorflow_models.official.benchmark.models import resnet_cifar_model
+from tensorflow_models.official.vision.image_classification.resnet import common as img_class_common
+
+flags.DEFINE_string("checkpoint_dir", "gs://cifar10_ckpt/",
+                    "Directory for writing model checkpoints.")
+flags.DEFINE_string("data_dir", "gs://cifar10_data/",
+                    "Directory for Resnet Cifar model input. Follow the "
+                    "instruction here to get Cifar10 data: "
+                    "https://github.com/tensorflow/models/tree/r1.13.0/official/resnet#cifar-10")
+flags.DEFINE_string("train_log_dir", "gs://cifar10_train_log/",
+                    "Directory for Resnet Cifar training logs")
+flags.DEFINE_boolean(
+    "use_in_process_cluster", False,
+    "Whether to use in-process cluster for testing.")
+flags.DEFINE_boolean(
+    "run_in_process_training", True,
+    "Whether to use in-process cluster to run training or evaluation.")
+
+FLAGS = flags.FLAGS
+
+TRAIN_EPOCHS = 182
+STEPS_PER_EPOCH = 781
+BATCH_SIZE = 64
+EVAL_BATCH_SIZE = 8
+
+def create_in_process_cluster(num_workers, num_ps):
```

Comment:
Could you update the work_config part according to this tutorial? https://www.tensorflow.org/tutorials/distribute/parameter_server_training#in-process_cluster

Reply:
Added inter ops.